core/state/statedb.go

```diff
@@ -1043,3 +1043,21 @@ func (s *StateDB) AddressInAccessList(addr common.Address) bool {
 func (s *StateDB) SlotInAccessList(addr common.Address, slot common.Hash) (addressPresent bool, slotPresent bool) {
 	return s.accessList.Contains(addr, slot)
 }
+
+// AccessList returns a list of dirty slots and addresses
```

Comment:
So, this will only ever give you the dirty storage, i.e those which have been modified during the execution. It won't give you the pure `SLOAD`:ed ones. 

An alternative way to implement this feature would be to, iteratively 
- exec the tx, then dump out the access-list which has now been filled with reads and writes (plus precompiles) 
- exec the tx again, this time with the access list as input. Dump out the new one. If it's changed (we reached some new code path), goto 1 
- Dump out the access list (minus precompiles, from and to)